src/jit/codegenxarch.cpp

```diff
@@ -6827,6 +6860,12 @@ void        CodeGen::genCompareLong(GenTreePtr  treeNode)
     assert(varTypeIsLong(op1->TypeGet()) && varTypeIsLong(op2->TypeGet()));
```

Comment:
Do we also need to assert the following before line 6860?

assert(treeNode->OperIsCompare());